geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/JoinProcess.scala

```diff
@@ -0,0 +1,121 @@
+import org.geotools.feature.DefaultFeatureCollection
+import org.geotools.process.ProcessException
+import org.geotools.process.factory.{DescribeParameter, DescribeProcess, DescribeResult}
+import org.geotools.process.vector.VectorProcess
+import org.geotools.util.NullProgressListener
+import org.locationtech.geomesa.accumulo.process.query.{QueryResult, QueryVisitor}
+import org.locationtech.geomesa.accumulo.process.unique.UniqueProcess
+import org.locationtech.geomesa.accumulo.util.SelfClosingIterator
+import org.locationtech.geomesa.filter.function.BinaryOutputEncoder
+import org.locationtech.geomesa.filter.function.BinaryOutputEncoder.EncodingOptions
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.opengis.util.ProgressListener
+
+/**
+  * Returns features from a feature type based on a join against a second feature type.
+  */
+@DescribeProcess(
+  title = "Join Process",
+  description = "Queries a feature type based on attributes from a second feature type"
+)
+class JoinProcess extends VectorProcess with LazyLogging {
+
+  // TODO this is currently aimed at a 1.2.2 backport - instead of the UniqueProcess we could run a stats job
+  /**
+    * @param unique result of running UniqueProcess
+    * @param join joined data set from which return features will be taken
+    * @param attribute attribute to join on
+    * @param bins flag to return results in BIN format
+    * @param binDtg date field for bin records - will use default date if not specified
+    * @param binTrackId track ID field for bin records - will not include trackId if not specified
+    * @param binLabel label field for bin record (optional)
+    * @param monitor listener to monitor progress
+    * @throws org.geotools.process.ProcessException if something goes wrong
+    * @return
+    */
+  @throws(classOf[ProcessException])
+  @DescribeResult(name = "result", description = "Output features")
+  def execute(@DescribeParameter(name = "unique", description = "Result of unique process for joined attribute")
+              unique: SimpleFeatureCollection,
+              @DescribeParameter(name = "join", description = "Joined features")
+              join: SimpleFeatureCollection,
+              @DescribeParameter(name = "attribute", description = "Attribute field to join on", min = 1)
+              attribute: String,
+              @DescribeParameter(name = "bins", description = "Return BIN records or regular records", min = 0)
+              bins: java.lang.Boolean,
+              @DescribeParameter(name = "binDtg", description = "Date field to use for BIN records", min = 0)
+              binDtg: String,
+              @DescribeParameter(name = "binTrackId", description = "Track field to use for BIN records", min = 0)
+              binTrackId: String,
+              @DescribeParameter(name = "binLabel", description = "Label field to use for BIN records", min = 0)
+              binLabel: String,
```

Comment:
1. the bin arguments are optional (that's the min = 0), but I couldn't find a good way to pass then in other than this. The WPS output formats don't have any mechanism for passing information in, so I had to hack it with a static object.
2. I could see how that would be useful. Not sure it's a use-case at the moment though... I'll try to add it anyway.